Python/_warnings.c

```diff
@@ -431,6 +431,45 @@ call_show_warning(PyObject *category, PyObject *text, PyObject *message,
     return -1;
 }
 
+static int
+tb_displayline(PyObject *f, PyObject *filename, int lineno, PyObject *name)
+{
+    int err;
+    PyObject *line;
+
+    if (filename == NULL || name == NULL)
+        return -1;
+    line = PyUnicode_FromFormat("  File \"%U\", line %d, in %U\n",
+                                filename, lineno, name);
+    if (line == NULL)
+        return -1;
+    err = PyFile_WriteObject(line, f, Py_PRINT_RAW);
+    Py_DECREF(line);
+    if (err != 0)
+        return err;
+    /* ignore errors since we can't report them, can we? */
+    if (_Py_DisplaySourceLine(f, filename, lineno, 4))
+        PyErr_Clear();
+    return err;
+}
+
+
+static void
+print_frames(PyFrameObject *frame)
```

Comment:
I suggest to rename it to print_stack() to be more consistent with the traceback module.

Reply:
Also printing correctly in C and having the exact same behavior in warnigns.py and warning.c seem a hassle, can I go the PyImport route and do the equivalent of
```
f = sys._getframe().f_back.f_back
traceback.print_stack(f=f)
```

I know that warning.c is meant to be efficient but here it's an opt-in path, so maybe not that necessary to be completely in C ?